### PR TITLE
Remove legacy flannel integration

### PR DESCRIPTION
**Pull request by tgraf**
The old flannel integration is not really being used. Chaining on top of
flannel is also possible with the generic-veth chaining method.
Deprecate the legacy flannel integration and use the standard
generic-veth chaining instead.

### CODE REVIEW

**Review by aanm — APPROVED** [review]

File: pkg/datapath/linux/config/config.go [hunk 1]
```diff
@@ -595,9 +595,6 @@ is_l3; })`))
 
 func (h *HeaderfileWriter) writeNetdevConfig(w io.Writer, cfg datapath.DeviceConfiguration) {
 	fmt.Fprint(w, cfg.GetOptions().GetFmtList())
-	if option.Config.IsFlannelMasterDeviceSet() {
-		fmt.Fprint(w, "#define HOST_REDIRECT_TO_INGRESS 1\n")
```
Comment (brb) [inline pkg/datapath/linux/config/config.go:599]: I think the legacy flannel integration was the only user of `HOST_REDIRECT_TO_INGRESS`, wasn't it (/cc @aanm)? If yes, then we should remove all `#ifdef HOST_REDIRECT_TO_INGRESS` in the datapath.

**Comment by brb** [comment]
Oh wow, this will close many issues :partying_face:

**Review by kaworu — APPROVED** [review]
Helm related changes LGTM. Seems to me that the PR is technically more a removal rather than deprecation. If correct, I think the release-note should make it clear.

**Comment by tgraf** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by gandro — APPROVED** [review]

**Comment by pchaigno** [comment]
test-gke

**Comment by joestringer** [comment]
Jenkins-based GKE runs are currently broken due to infrastructure issue. CI 3.0 GKE was successful. We'll get plenty of subsequent testing as we branch for v1.10.0-rc1 so not too worried about bypassing this failure to merge. The other failures are linked to lack of rebase per @pchaigno's comment above.

Merging.

**Pull request merged**